system-addon/content-src/components/Base/Base.jsx

```diff
@@ -22,13 +23,17 @@ function addLocaleDataForReactIntl({locale}) {
 
 class Base extends React.Component {
   componentDidMount() {
+    this.sendNewTabRenderedEvent(this.props.App);
```

Comment:
Maybe rename this function to `sendNewTabRehydrated`.

Also, you should probably use either `componentWillMount` + `componentWillUpdate` OR `componentDidMount` + `compontentDidUpdate` to be consistent with the timing